pages/agent/v3/hooks.md.erb

```diff
@@ -1,40 +1,69 @@
+Agent hooks can be used to extend or override the built-in behaviours of the
+Buildkite Agent and Bootstrap binaries.
 
-{:toc}
-
-## Types of hooks
-
-There are three types of hooks:
+Hooks can be defined globally on your agent hosts, in your repository, or from
+[Plugins](/docs/plugins). For example, you could create a global `checkout` hook
+which speeds up a fresh `git clone` on a new build machine, a repository
+`pre-command` hook which sets up repository-specific environment variables, or a
+plugin `environment` hook which exports secret API keys as environment
+variables.
 
-* Agent hooks - these exist on the agent’s machine, and are run for every job on any pipeline.
-* Repository hooks - these exist in your pipeline repository’s `.buildkite/hooks` directory.
-* Plugin hooks - these are provided by any [plugins](/docs/plugins) you've defined in your `pipeline.yml`.
+{:toc}
 
-### Agent hooks
+## Categories of hooks
 
-Each agent installer comes with a hooks directory containing a set of sample hooks. You can find the location of your agent hooks directory in your platform’s installation documentation.
+There are two categories of hook:
 
-To get started with agent hooks copy the relevant example script and remove the `.sample` file extension.
+* Lifecycle hooks
+* Job hooks
 
-### Repository hooks
+Lifecycle hooks are **executed** by the Buildkite Agent as part of the agent
+lifecycle. For example, the `pre-bootstrap` hook is executed before starting a
+per-job bootstrap process, and the `agent-shutdown` hook is executed before the
+agent process terminates.
 
-Repository hooks allow you to execute repository-specific scripts. Repository hooks live alongside your repository’s source code under the directory `.buildkite/hooks`.
+Job hooks are **sourced** by the Buildkite Bootstrap in the different job
+phases. They are run in a per-job shell environment, and any exported
+environment variables are carried to the job’s subsequent phases and hooks. For
+example, the `environment` hook can modify or export new environment variables
+for the job’s subsequent phases and hooks.
 
-To get started, create a shell script in `.buildkite/hooks` named `post-checkout`.
-It will be sourced and run every build, after the Agent checks your repository out.
-You can also use one of the other available *Repository* hooks, according to when you want your custom script to run.
+## Lifecycle hooks
 
-### Plugin hooks
-
-Plugin hooks allow plugins you’ve defined in your Pipeline to override default behavior.
-
-See the [plugin documentation](/docs/plugins) for how to implement plugin hooks.
+<table>
```

Comment:
I work err on the side of including a mostly redundant "Order" column in this table, to make it clear that both are Global hooks, in the same format as the Job hook table.

Reply:
I spent a while thinking about this and I think it’s the right call :+1: What do you think about naming both columns "Location Order"? I’m going to make that change and we can always revert it 😄

Reply:
I actually wanted to redesign that table a few months ago, but haven't had any design time. Leave it as location order and we'll reexamine later?